torch/csrc/jit/runtime/static/impl.h

```diff
@@ -151,6 +163,25 @@ class TORCH_API StaticRuntime {
     DCHECK(i < module_->output_regs.size());
     return reg_[module_->output_regs[i]];
   }
+  friend class MemoryPlanner;
+};
+
+class MemoryPlanner {
+ public:
+  MemoryPlanner(StaticRuntime* runtime);
+
+  void allocate_memory();
+  void clean_up_memory(std::vector<IValue>& reg);
+
+ private:
+  std::unordered_set<size_t> reg_out_variant_;
+  std::vector<c10::StorageImpl*> internal_blob_storages_;
```

Comment:
nit: `internal_storages_`? just since `blob` is an ever-so-slightly different concept and might be confusing